src/python/dart/worker/engine.py

```diff
@@ -111,37 +109,61 @@ def _transition_queued_actions_to_pending(self):
                 continue
 
             except Exception as e:
-                _logger.error('error transitioning action (id=%s) to PENDING: %s' % (action.id, e.message))
+                error_message = e.message + '\n\n\n' + traceback.format_exc()
+                _logger.error('error transitioning action (id=%s) to PENDING: %s' % (action.id, error_message))
 
             finally:
                 db.session.rollback()
 
+    def _get_latest_active_job_definition(self, job_def_name):
```

Comment:
Are there any implications of using this in terms of workflows getting version changed in the middle of running them? e.g. step 1 is redshift_engine_prd:2 but step 2 is run on redshift_engine_prd:3?